### PR TITLE
Refactor CI workflow

### PR DESCRIPTION
**Pull request by sczyh30**
Signed-off-by: Eric Zhao <sczyh16@gmail.com>

### Describe what this PR does / why we need it

Refactor CI workflow and split CI into two parts with golangci-lint-action.

Update go version to 1.16 in `go.mod`.

### Does this pull request fix one issue?

Resloves #499 

### Describe how to verify it

The CI should pass.

### Special notes for reviews

NONE

### CODE REVIEW

**Review by jnan806 — COMMENTED** [review]
image
I think can *_test.go can be ignored, the ignore setting is:
image

**Review by jnan806 — COMMENTED** [review]
image
If you want to ignore this error, you can set like: 
image

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Options for analysis running.
+run:
+  # vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
+  skip-dirs-use-default: true
+  skip-dirs:
+    - example
```
Comment (jnan806) [inline .golangci.yml:6]: recommends to add `-test`

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  skip-files:
+    - ".*\\.pb\\.go$"
```
Comment (jnan806) [inline .golangci.yml:8]: recommends to add `- _test.go`

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Options for analysis running.
+run:
+  # vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
```
Comment (jnan806) [inline .golangci.yml:3]: recommends to add 
``` yaml  
# Include test files or not.
# Default: true
tests: false
```

image

**Comment by jnan806** [comment]
following errors occurred in comments, I have no idea to fix it now.

image

**Review by binbin0325 — COMMENTED** [review]
fix lint check ， delete core/hotspot/rule.go#104

**Pull request merged**